docs/rules/callback-return.md

```diff
@@ -0,0 +1,121 @@
+# Enforce Return After Callback (callback-return)
+
+The callback pattern is at the heart most I/O and event programming
+ in JavaScript.
+
+```js
+function doSomething(err, callback) {
+    if (err) {
+        return callback(err);
+    }
+    callback();
+}
+```
+
+To prevent calling the callback multiple times it's important to `return` anytime the callback is triggered outside
+ of the main function body. Neglecting this technique often leads to issues where you do something more than once.
+ For example in the case of an HTTP request, you may try to send HTTP headers more than once leading node.js to `throw`
+ a `Can't render headers after they are sent to the client.` error.
+
+## Rule Details
+
+This rule is aimed at ensuring that callbacks used outside of the main function block are always part-of or immediately
+preceding a `return` statement. This rules decides what is a callback based on the name of the function being called.
+By default the rule treats `cb`, `callback`, and `next` as callbacks.
+
+The following patterns are considered warnings:
+
+```js
+function foo() {
+    if (err) {
+        callback(err);
+    }
+    callback();
+}
+```
+
+The following patterns are not considered warnings:
+
+```js
+function foo() {
+    if (err) {
+        return callback(err);
+    }
+    callback();
+}
+```
+
+### Options
+
+The rule takes a single option, which is an array of possible callback names.
+
+```json
+callback-return: [2, ["callback", "cb", "next"]]
+```
+
+### Gotchas
+
+There are many cases of bad behavior that this rule will not catch and even a few cases where you're doing the right
+thing but this rule will warn, primarily because of the difficulty of static analyzing intent.
```

Comment:
The ending is hard to understand. Maybe "difficulty of determining intent through static code analysis"?